### PR TITLE
Feature/update nextjs13

### PR DESCRIPTION
**Pull request by luas10c**
### Features
* add support nextjs13
* add eslint
* add prettier
* add commitlint
* add sharp resize image in production mode
* update framer-motion
* update @chakra-ui/react
* update @chakra-ui/icons
* update husky v8
* add lint-staged
* add husky hooks commit-msg, pre-commit and pre-push
* fix more bugs
* fix layout shift caused by fonts

### CODE REVIEW

**Comment by fernandounger** [comment]
nice PR🎉
I was waiting for this nextjs update

**Comment by fpdsjr** [comment]
Wow insane job, you should be proud of this work gj!

**Comment by hauptrolle** [comment]
@luas10c Nice! I will try to take a look at this this week!

**Review by hauptrolle — CHANGES_REQUESTED** [review]
What a PR 🙈 Looks good so far, just some small config changes.

File: .prettierrc.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/** @type{import('prettier').Config} */
+module.exports = {
+  tabWidth: 2,
+  useTabs: false,
+  singleQuote: true,
+  jsxBracketSameLine: true,
```
[the remark this answers is not in the document]
Comment (luas10c) [reply .prettierrc.js:6]: this two?

singleQuote: true,
jsxBracketSameLine: true,

[the remark this answers is not in the document]
Comment (luas10c) [reply .prettierrc.js:6]: I didn't change that, it's like the original, just compare.

**Comment by luas10c** [comment]
@hauptrolle 

```js
// .eslintrc.js

/** @type{import('prettier').Config} */
module.exports = {
  tabWidth: 2,
  useTabs: false,
  singleQuote: true,
  jsxBracketSameLine: true,
  bracketSpacing: true,
  trailingComma: 'all',
  arrowParens: 'always',
  singleAttributePerLine: false,
  endOfLine: 'lf',
  bracketSameLine: true,
  printWidth: 90,
  semi: false,
  plugins: []
}
```

Also removed this on body:
<body cz-shortcut-listen="true">

This was from an extension called colorzilla

**Comment by hauptrolle** [comment]
Merged!

**Pull request merged**